pokemongo_bot/cell_workers/evolve_pokemon.py

```diff
@@ -65,19 +70,38 @@ def work(self):
                 candy = inventory.candies().get(pokemon.pokemon_id)
                 pokemon_to_be_evolved = pokemon_to_be_evolved + min(candy.quantity / (pokemon.evolution_cost - 1), filtered_dict[pokemon.pokemon_id])
 
-        if pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved:
+        self._print_check(pokemon_to_be_evolved)
+
+        has_minimum_to_evolve = pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved
+        if has_minimum_to_evolve:
             if self.use_lucky_egg:
                 self._use_lucky_egg()
             cache = {}
             for pokemon in filtered_list:
                 if pokemon.can_evolve_now():
                     self._execute_pokemon_evolve(pokemon, cache)
 
+    def _print_check(self, pokemon_to_be_evolved):
+        has_minimum_to_evolve = pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved
+        result_message = ("Gotta catch`em all!", "Gotta evolv`em all!")[has_minimum_to_evolve]
+        self.emit_event(
+            'pokemon_evolve_check',
+            formatted='Checking... Has {has}, needs {needs}. {message}',
```

Comment:
I think it can be even more simplistic: `Evolvable: {has}/{need}` will be sufficient. 👍